src/legacy/core_plugins/vis_type_timeseries/public/visualizations/views/timeseries/index.js

```diff
@@ -136,79 +136,80 @@ export const TimeSeries = ({
+                  stackAsPercentage={isPercentage}
+                  xScaleType={xScaleType}
+                  yScaleType={yScaleType}
+                  timeZone={timeZone}
+                  enableHistogramMode={enableHistogramMode}
+                  useDefaultGroupDomain={useDefaultGroupDomain}
+                  sortIndex={sortIndex}
+                />
+              );
+            }
 
-          if (bars.show) {
-            return (
-              <BarSeriesDecorator
-                key={key}
-                seriesId={id}
-                seriesGroupId={groupId}
-                name={label.toString()}
-                data={data}
-                hideInLegend={hideInLegend}
-                bars={bars}
-                color={color}
-                stackAccessors={stackAccessors}
-                stackAsPercentage={isPercentage}
-                xScaleType={xScaleType}
-                yScaleType={yScaleType}
-                timeZone={timeZone}
-                enableHistogramMode={enableHistogramMode}
-                useDefaultGroupDomain={useDefaultGroupDomain}
-                sortIndex={sortIndex}
-              />
-            );
-          }
+            if (lines.show) {
+              return (
+                <AreaSeriesDecorator
+                  key={key}
+                  seriesId={id}
+                  seriesGroupId={groupId}
+                  name={label.toString()}
+                  data={data}
+                  hideInLegend={hideInLegend}
+                  lines={lines}
+                  color={color}
+                  stackAccessors={stackAccessors}
+                  stackAsPercentage={isPercentage}
+                  points={points}
+                  xScaleType={xScaleType}
+                  yScaleType={yScaleType}
+                  timeZone={timeZone}
+                  enableHistogramMode={enableHistogramMode}
+                  useDefaultGroupDomain={useDefaultGroupDomain}
+                  sortIndex={sortIndex}
+                />
+              );
+            }
 
-          if (lines.show) {
-            return (
-              <AreaSeriesDecorator
-                key={key}
-                seriesId={id}
-                seriesGroupId={groupId}
-                name={label.toString()}
-                data={data}
-                hideInLegend={hideInLegend}
-                lines={lines}
-                color={color}
-                stackAccessors={stackAccessors}
-                stackAsPercentage={isPercentage}
-                points={points}
-                xScaleType={xScaleType}
-                yScaleType={yScaleType}
-                timeZone={timeZone}
-                enableHistogramMode={enableHistogramMode}
-                useDefaultGroupDomain={useDefaultGroupDomain}
-                sortIndex={sortIndex}
-              />
-            );
+            return null;
           }
-
-          return null;
-        }
-      )}
+        )
+        .reverse()}
```

Comment:
Let's add a comment here, why we should revert the array here.
This is not so obvious, and definitely in the future someone will try to remove it.